bots/sg_rules_delete_by_scope.py

```diff
@@ -0,0 +1,134 @@
+"""
+## sg_rules_delete_by_scope
+What it does: Deletes all rules on a security group with a specific scope, port and protocol are optional
+Usage: AUTO: sg_rules_delete_by_scope <scope> <direction> <port|*> <protocol|*>
+
+Example: AUTO: sg_rules_delete_by_scope 0.0.0.0/0 inbound 22 tcp
+Parameters:
+    scope: a.b.c.d/e
+    direction: inbound/ outbound
+    port: number/ *
+    protocol: TCP/ UDP/ *
+-When '*' is any value of the parameter
+
+Other Examples:
+    all rules with 1.0.0.0/16 scope will be deleted for any port and protocol:
+    sg_rules_delete_by_scope 1.0.0.0/16 inbound * *
+
+    all rules with 0.0.0.0/0 scope will be deleted for port 22 and any protocol:
+    sg_rules_delete_by_scope 0.0.0.0/0 inbound 22 *
+Notes :
+    - the bot deletes the rule without splitting ports ( do not create new rules without the deleted port)
+      for deleting rule with split use - sg_single_rule_delete bot .
+    -  before running this bot, ensure that your applications will work correctly without those rules
+    - if a port is in a port range the rule wont be deleted ! use * on port parameter to delete the rule for any port
+Limitations: IPv6 is not supported
+
+"""
+
+import re
+
+PORT_TO = 'portTo'
+PORT_FROM = 'port'
+PROTOCOL = 'protocol'
+SCOPE = 'scope'
+ALL_TRAFFIC_PORT = 0
+ALL_TRAFFIC_PROTOCOL = '-1'
+
+"""
+checks for ip validity ,else fix it to be right
+"""
+
+
+def verify_scope_is_cidr(rule):
+    ip = re.split('/|\.', str(rule[SCOPE]))  # break ip to blocks
+    rule[SCOPE] = ip[0] + '.' + ip[1] + '.' + ip[2] + '.' + ip[3] + '/' + ip[4]
+    pass
+
+
+"""
+returns a string of rule's id by scope,port,direction,etc.
+"""
+
+
+def stringify_rule(rule):
```

Comment:
I believe this function can be used in multiple bots, you can create utils file with scripts that can be shared between bots. You can do it in different branch but you can start think about generalise some of the methods

Reply:
I will make another bot_utils file in another branch and move all of the functions to one place